main.go

```diff
@@ -0,0 +1,605 @@
+	}
+	return results, nil
+}
+
+// eth_gasPrice (returns the gas price)
+// GasPrice returns a suggestion for a gas price for legacy transactions.
+func (s *BlockChainAPI) GasPrice(ctx context.Context) (*hexutil.Big, error) {
+	return (*hexutil.Big)(big.NewInt(10102020506)), nil
+}
+
+// eth_maxPriorityFeePerGas
+// MaxPriorityFeePerGas returns a suggestion for a gas tip cap for dynamic fee transactions.
+func (s *BlockChainAPI) MaxPriorityFeePerGas(ctx context.Context) (*hexutil.Big, error) {
+	return (*hexutil.Big)(big.NewInt(10102020506)), nil
+}
+
+// eth_getBalance (returns the balance for any given block)
+// GetBalance returns the amount of wei for the given address in the state of the
+// given block number. The rpc.LatestBlockNumber and rpc.PendingBlockNumber meta
+// block numbers are also allowed.
+func (s *BlockChainAPI) GetBalance(
+	ctx context.Context,
+	address common.Address,
+	blockNrOrHash *rpc.BlockNumberOrHash,
+) (*hexutil.Big, error) {
+	return (*hexutil.Big)(big.NewInt(101)), nil
+}
+
+// eth_getCode (returns the code for the given address)
+// GetCode returns the code stored at the given address in the state for the given block number.
+func (s *BlockChainAPI) GetCode(
+	ctx context.Context,
+	address common.Address,
+	blockNrOrHash rpc.BlockNumberOrHash,
+) (hexutil.Bytes, error) {
+	return hexutil.Bytes{}, nil
+}
+
+// Result structs for GetProof
+type AccountResult struct {
+	Address      common.Address  `json:"address"`
+	AccountProof []string        `json:"accountProof"`
+	Balance      *hexutil.Big    `json:"balance"`
+	CodeHash     common.Hash     `json:"codeHash"`
+	Nonce        hexutil.Uint64  `json:"nonce"`
+	StorageHash  common.Hash     `json:"storageHash"`
+	StorageProof []StorageResult `json:"storageProof"`
+}
+
+type StorageResult struct {
+	Key   string       `json:"key"`
+	Value *hexutil.Big `json:"value"`
+	Proof []string     `json:"proof"`
+}
+
+// eth_getProof (returns state proof for an account)
+// GetProof returns the Merkle-proof for a given account and optionally some storage keys.
+func (s *BlockChainAPI) GetProof(
+	ctx context.Context,
+	address common.Address,
+	storageKeys []string,
+	blockNrOrHash rpc.BlockNumberOrHash,
+) (*AccountResult, error) {
+	storageProof := make([]StorageResult, len(storageKeys))
+	return &AccountResult{
+		Address:      address,
+		AccountProof: []string{""},
+		Balance:      (*hexutil.Big)(big.NewInt(10011)),
+		CodeHash:     common.Hash{},
+		Nonce:        hexutil.Uint64(10502),
+		StorageHash:  common.Hash{},
+		StorageProof: storageProof,
+	}, nil
+}
+
+// eth_getStorageAt (we probably won't support this day 1)
+// GetStorageAt returns the storage from the state at the given address, key and
+// block number. The rpc.LatestBlockNumber and rpc.PendingBlockNumber meta block
+// numbers are also allowed.
+func (s *BlockChainAPI) GetStorageAt(
+	ctx context.Context,
+	address common.Address,
+	hexKey string, blockNrOrHash rpc.BlockNumberOrHash,
```

Comment:
What is `hexKey`, maybe `publicKey`? also might be better to not put `hex` in the name, since you will validate the input anyway and you can detect the encoding. You should put that in the comment.

Reply:
Good point :+1:  In the specification it is written:
`Returns the value from a storage position at a given address.` So `hexKey` is actually the storage position.
And the accepted value is a `hex encoded 256 bit unsigned integer. Most of the naming comes from the `go-ethereum` repository, but I will rename the arguments accordingly, to make them more readable.

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/10/commits/71ef6d187aed9fd0ea0b73b59cfd2a4829a0d569